modules/caddyhttp/caddyauth/caddyauth.go

```diff
@@ -70,6 +74,10 @@ func (a Authentication) ServeHTTP(w http.ResponseWriter, r *http.Request, next c
 			log.Printf("[ERROR] Authenticating with %s: %v", provName, err)
 			continue
 		}
+		a.logger.Debug("failed authentication",
```

Comment:
Shouldn't this be checking `authed`? Currently it appears to be logging this message for every authentication attempt regardless of success/failure.

Also, it doesn't contain the username.

Here is an example;

```
{
  "level": "debug",
  "ts": 1588821522.261988,
  "logger": "http.handlers.authentication",
  "msg": "failed authentication",
  "provider": "http_basic",
  "request": {
    "method": "GET",
    "uri": "/service/",
    "proto": "HTTP/1.1",
    "remote_addr": "x.x.x.x:20454",
    "host": "xxx.nz",
    "headers": {
      "Connection": [
        "Keep-Alive"
      ],
      "Authorization": [
        "Basic xxx"
      ]
    },
    "tls": {
      "resumed": false,
      "version": 772,
      "ciphersuite": 4865,
      "proto": "",
      "proto_mutual": true,
      "server_name": "xxx.nz"
    }
  }
}
```

Reply:
Oh, yep. I'll fix that.

As for the username, you didn't say you need that directly. You said you wanted to know if any credentials were given. Since basicauth is only one of N provider modules that all work differently, we just log the request and you can look for what you need to look for. In your case, a non-empty Authorization header.